packages/e2e-playwright/tests/e2e/card/kcpAuthentication.spec.ts

```diff
@@ -1,27 +1,46 @@
+import { expect, test } from '../../../fixtures/card.fixture';
+import { URL_MAP } from '../../../fixtures/URL_MAP';
+import {
+    KOREAN_TEST_CARD,
+    PAYMENT_RESULT,
+    REGULAR_TEST_CARD,
+    TEST_CVC_VALUE,
+    TEST_DATE_VALUE,
+    TEST_PWD_VALUE,
+    TEST_TAX_NUMBER_VALUE
+} from '../../utils/constants';
 
 test.describe('Card payments with KCP enabled feature', () => {
-    test('should submit the korea issue card payment', async () => {
-        // fill in KOREAN_TEST_CARD
-        // expect to see the non-branded logo
-        // fill in the required fields
-        // click pay btn
-        // expect to see success msg
+    test('should submit the korea issue card payment', async ({ cardWithKCP }) => {
+        await cardWithKCP.goto(URL_MAP.cardWithKcp);
+        await cardWithKCP.fillCardNumber(KOREAN_TEST_CARD);
+        await cardWithKCP.cardNumberKoreanBrand.waitFor({ state: 'visible' });
+        await cardWithKCP.fillCvc(TEST_CVC_VALUE);
+        await cardWithKCP.fillExpiryDate(TEST_DATE_VALUE);
+        await cardWithKCP.taxNumberInput.fill(TEST_TAX_NUMBER_VALUE);
+        await cardWithKCP.passwordInput.fill(TEST_PWD_VALUE);
+        await cardWithKCP.pay();
+        await expect(cardWithKCP.paymentResult).toContainText(PAYMENT_RESULT.authorised);
     });
 
-    test('should not submit the korea issue card payment', async () => {
-        // fill in KOREAN_TEST_CARD
-        // expect to see the non-branded logo
-        // fill in partial fields
-        // click pay btn
-        // expect to see error msg
+    test('should not submit the korea issue card payment', async ({ cardWithKCP }) => {
+        await cardWithKCP.goto(URL_MAP.cardWithKcp);
+        await cardWithKCP.fillCardNumber(KOREAN_TEST_CARD);
+        await cardWithKCP.cardNumberKoreanBrand.waitFor({ state: 'visible' });
+        await cardWithKCP.fillCvc(TEST_CVC_VALUE);
+        await cardWithKCP.fillExpiryDate(TEST_DATE_VALUE);
+        await cardWithKCP.pay();
+        await expect(cardWithKCP.taxNumberErrorLocator).toContainText('Invalid Cardholder birthdate or Corporate registration number');
+        await expect(cardWithKCP.passwordErrorLocator).toContainText('Enter the password');
     });
 
-    test('should submit the regular non-korean card payment', async () => {
-        // fill in NON_KOREAN_TEST_CARD
-        // expect to see the brand logo
-        // fill in required fields
-        // click pay btn
-        // expect to see success msg
+    test('should submit the regular non-korean card payment', async ({ cardWithKCP, page }) => {
+        await cardWithKCP.goto(URL_MAP.cardWithKcp);
+        await cardWithKCP.fillCardNumber(REGULAR_TEST_CARD);
+        await cardWithKCP.fillCvc(TEST_CVC_VALUE);
+        await cardWithKCP.fillExpiryDate(TEST_DATE_VALUE);
+        await cardWithKCP.isPayable();
+        await cardWithKCP.pay();
+        await expect(cardWithKCP.paymentResult).toContainText(PAYMENT_RESULT.authorised);
```

Comment:
Does this  whole file not overlap with `e2e/card/kcp/card.kcp.spec.ts`?

Reply:
I am not 100% sure because there is a `paymentSuccessfulMock` in the `e2e/card/kcp/card.kcp.spec.ts` file i suppose it's not really making a payment call.
I think if we remove the `paymentSuccessfulMock` those 2 files are overlapping.

Reply:
Yeah, I think, at the time, we weren't sure if we had an actual, working korean local card number - so I guess that's why @ribeiroguilherme used a mock?